src/Bounds.cpp

```diff
@@ -1235,8 +1235,26 @@ class Bounds : public IRVisitor {
                                        !b_interval.min.type().is_uint() &&
```

Comment:
I believe that this should be the case, but @abadams might contradict me. I have been operating under the assumption that that invariant is maintained.

Reply:
(Would be nice to have that documented somewhere, though that's orthogonal to this PR.)

I ask mainly because code lower down seemed to be optimized based on that assumption, but it wasn't clear to me whether deliberate or not.

Reply:
I agree, all of the changes I have made to bounds inference assumes that the type of the min, type of the max, and type of the operation are all equal.